src-extras/Database/LSMTree/Extras/Index.hs

```diff
@@ -0,0 +1,64 @@
+module Database.LSMTree.Extras.Index
+(
+    Append (AppendSinglePage, AppendMultiPage),
+    append,
+    append'
+)
+where
+
+import           Control.DeepSeq (NFData (rnf))
+import           Control.Monad.ST.Strict (ST)
+import           Data.Foldable (toList)
+import           Data.Word (Word32)
+import           Database.LSMTree.Internal.Chunk (Chunk)
+import           Database.LSMTree.Internal.IndexCompactAcc (IndexCompactAcc)
+import qualified Database.LSMTree.Internal.IndexCompactAcc as IndexCompact
+                     (appendMulti, appendSingle)
+import           Database.LSMTree.Internal.IndexOrdinaryAcc (IndexOrdinaryAcc)
+import qualified Database.LSMTree.Internal.IndexOrdinaryAcc as IndexOrdinary
+                     (appendMulti, appendSingle)
+import           Database.LSMTree.Internal.Serialise (SerialisedKey)
+
+-- | Instruction for appending pages, to be used in conjunction with indexes.
+data Append
+    = AppendSinglePage
+          SerialisedKey -- ^ Minimum key
+          SerialisedKey -- ^ Maximum key
+      {-^
+          Append a single page that fully comprises one or more key–value pairs.
+      -}
+    | AppendMultiPage
+          SerialisedKey -- ^ Sole key
+          Word32        -- ^ Number of overflow pages
+      {-^
+          Append multiple pages that together comprise a single key–value pair.
+      -}
```

Comment:
The suffix documentation for constructors isn't proper haddocks. Put the docs in front instead

```suggestion
-- | Instruction for appending pages, to be used in conjunction with indexes.
data Append
    = {- |
          Append a single page that fully comprises one or more key–value pairs.
      -}
      AppendSinglePage
          SerialisedKey -- ^ Minimum key
          SerialisedKey -- ^ Maximum key
    | {- |
          Append multiple pages that together comprise a single key–value pair.
      -}
      AppendMultiPage
          SerialisedKey -- ^ Sole key
          Word32        -- ^ Number of overflow pages
```

Reply:
Actually, [suffix documentation of data constructors *is* possible](https://haskell-haddock.readthedocs.io/latest/markup.html#constructors-and-record-fields), but the documentation must come right after the constructors instead of the whole data type alternatives. Looking at this now, it is quite obvious to me that what I did can’t work, as it is not clear whether documentation following a data type alternative with at least one field refers to the data constructor or the last field.

Reply:
Fixed in 9e8ec9e333c246f162ca7f2a73e385c2077ec09a.